oktapam/resource_password_settings.go

```diff
@@ -0,0 +1,274 @@
+			},
+			attributes.CharacterOptions: {
+				Type:        schema.TypeList,
+				Required:    true,
+				MinItems:    1,
+				MaxItems:    1,
+				Description: descriptions.CharacterOptions,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						attributes.UpperCase: {
+							Type:        schema.TypeBool,
+							Required:    true,
+							Description: descriptions.CharacterOptionsUpperCase,
+						},
+						attributes.LowerCase: {
+							Type:        schema.TypeBool,
+							Required:    true,
+							Description: descriptions.CharacterOptionsLowerCase,
+						},
+						attributes.Digits: {
+							Type:        schema.TypeBool,
+							Required:    true,
+							Description: descriptions.CharacterOptionsDigits,
+						},
+						attributes.Punctuation: {
+							Type:        schema.TypeBool,
+							Required:    true,
+							Description: descriptions.CharacterOptionsPunctuation,
+						},
+					},
+				},
+			},
+		},
+		Importer: &schema.ResourceImporter{
+			StateContext: resourcePasswordSettingsReadImport,
+		},
+	}
+}
+
+func formatPasswordSettingsID(resourceGroupID string, projectID string) string {
+	// password settings don't have an identifier in itself and is really an attribute of a project.
+	// we manage it as a separate resource since it's lifecycle is somewhat separate from a project
+	return fmt.Sprintf("%s/%s", resourceGroupID, projectID)
+}
+
+func parsePasswordSettingsID(resourceID string) (string, string, error) {
+	split := strings.Split(resourceID, "/")
+	if len(split) != 2 {
+		return "", "", fmt.Errorf("expected format: <resource_group_id>/<project_id>, received: %s", resourceID)
+	}
+	return split[0], split[1], nil
+}
+
+func resourcePasswordSettingsCreate(ctx context.Context, d *schema.ResourceData, m any) diag.Diagnostics {
+	c := m.(client.OktaPAMClient)
+	resourceGroupID := d.Get(attributes.ResourceGroup).(string)
+	projectID := d.Get(attributes.Project).(string)
+
+	passwordSettings, diags := readPasswordSettingsFromResource(d)
+
+	if diags != nil {
+		return diags
+	}
+
+	if err := c.UpdatePasswordSettings(ctx, resourceGroupID, projectID, passwordSettings); err != nil {
+		return diag.FromErr(err)
+	}
+
+	d.SetId(formatPasswordSettingsID(resourceGroupID, projectID))
+
+	return resourcePasswordSettingsRead(ctx, d, m)
+}
+
+func resourcePasswordSettingsRead(ctx context.Context, d *schema.ResourceData, m any) diag.Diagnostics {
+	var diags diag.Diagnostics
+	c := m.(client.OktaPAMClient)
+	resourceGroupID := d.Get(attributes.ResourceGroup).(string)
+	projectID := d.Get(attributes.Project).(string)
+
+	passwordSettings, err := c.GetPasswordSettings(ctx, resourceGroupID, projectID)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	if passwordSettings != nil && len(passwordSettings.ManagedPrivilegedAccountsConfig) > 0 {
+		for k, v := range passwordSettings.ToResourceMap() {
+			if err := d.Set(k, v); err != nil {
+				diags = append(diags, diag.FromErr(err)...)
+			}
+		}
```

Comment:
Should this have set the Id here?

```
if diags == nil {
    d.SetId(formatPasswordSettingsID(resourceGroupID, projectID))
}
```

Reply:
Theoretically, it doesn't need to since a `Read` should already have an ID set.